pkg/ebpf/events_derived.go

```diff
@@ -69,6 +71,64 @@ func (t *Tracee) deriveEvent(event trace.Event) []trace.Event {
 		deriveFns = map[int32]deriveFn{
 			ContainerRemoveEventID: containerRemoveFn,
 		}
+	case PrintSyscallTableEventID:
+		detectHookedSyscallsFn := func(def EventDefinition) error {
+			syscallsAdresses, err := getEventArgUlongArrVal(&event, "syscalls_addresses")
+			if err != nil {
+				return fmt.Errorf("error parsing syscalls_numbers arg: %v", err)
+			}
+			hookedSyscallData := make([]HookedSyscallData, 0, 0)
+			for idx, syscallsAdress := range syscallsAdresses {
+				InTextSegment, err := t.kernelSymbols.TextSegmentContains(syscallsAdress)
+				if err != nil {
+					fmt.Errorf(err.Error())
+					continue
+				}
+				if !InTextSegment {
+					hookingFunction, err := t.kernelSymbols.GetSymbolByAddr(syscallsAdress)
```

Comment:
I think you should export the building of the `HookedSyscallData` to a sub-function